### ISSUE TITLE
Unable to process contracts after upgrade @metamask/sdk from 0.1.0 to >= 0.2.0 and other issues

### ISSUE THREAD

**Issue by ugened47** [MetaMask/metamask-sdk#108]
**Describe the bug**
After upgrading sdk from 0.1.0 some functionality stopped working, especially one that interacts with a contracts which worked prior with sdk 0.1.0 and metamask-mobile 6.2.0. SDK v0.1.0 stoped work with versions higher than metamask-mobile 6.2.0.
Interaction with contracts was ok and worked as expected, however after upgrading in't no longer working. In addition there are 

Not working contract:

https://github.com/MetaMask/metamask-sdk/assets/9214135/1f72381e-90df-4d26-b229-a1b1f3bc56ec

Here is also a screencapture of unstable work:

https://github.com/MetaMask/metamask-sdk/assets/9214135/237c2cf5-263d-4d32-b620-0ff50de8f2ca

**To Reproduce**

[reproduction repo](https://github.com/ugened47/mm-issue-repro)

**Expected behaviour**
Contract interation works and sdk works stable.

**Screenshots**

** Provide environment information**
- Platform (Eg. Web / React-Native / Node.js / Unity): React-Native
- MetaMask SDK Version: 0.2.4
- MetaMask Mobile Version: 6.4.0
- Browser (if relevant):

**Additional context**

**Comment by abretonc7s** [comment]
Hi @ugened47 , thank you for providing a repo. 
One of the main difference with the update to 0.2.x is to remove rn-nodeify which may cause your issue.
I will investigate your example and get back to you quickly.

**Comment by ugened47** [comment]
@abretonc7s hi. I'm not using rn-nodify. See the attached repo, please. It's using node-libs-react-native and based on the example from the metamask-sdk monorepo. Thank you.

**Comment by abretonc7s** [comment]
Yes i understand and have checked your example. It seems to be an issue with ethers.js when not using rn-nodeify. I will investigate and see how to provide a fix asap.

**Comment by ugened47** [comment]
@abretonc7s In app I'm working on, where 0.1.0 works well with older versions of metamask-mobile, I'm not using rn-nodify either. Using node-libs-react-native in conjunction with expo-crypto-polyfillls

**Comment by ugened47** [comment]
@TayfunCesur, it still doesn't work with the latest SDK and mobile app, regardless of whether I use ethers.js or web3.js.

**Comment by TayfunCesur** [comment]
Yes @ugened47 , I'm also facing an issue about interacting the contracts with new sdk (in my case is ERC20). I posted my comment [here](https://github.com/MetaMask/metamask-sdk/issues/144#issuecomment-1599074588). 
For the devs information, I have detected that the issue is related to the provider which comes from `sdk.getProvider()`

**Comment by christopherferreira9** [comment]
Hi @ugened47! We'll be looking at this as soon as possible.

**Comment by abretonc7s** [comment]
We are working on a react native kit and provide additional hooks to facilitate contract integration.
We will keep you updated next week after we have completed the feature and testing.
